File: content/code/factorial.R
#' Computes the factorial of n
#'
#' @param n The number to compute the factorial of.
#' @return The factorial of n
factorial <- function(n) {
  if (n < 0)
    stop('received negative input')
  if (n == 0)
    return(1)

  result <- 1
  for (i in 1:n)
    result <- result * i
  result
}

File: content/code/fortran/factorial.f90
module factorial_mod
contains
   ! computes the factorial of n
   integer function factorial(n)
      implicit none
      integer, intent(in) :: n
      integer r
      integer i
      if(n < 0) then
         write(*,*) 'Received negative input'
         stop
      end if
      r = 1
      do i = 1,n
         r = r*i
      end do
      factorial=r
   end function factorial
end module factorial_mod
